firmware-support/memmap-generate/src/generators/mod.rs

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2024 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+
+use device_instances::generate_device_instances_struct;
+use device_types::generate_device_type;
+use proc_macro2::{Ident, Span, TokenStream};
+use types::generate_type_def;
+
+use crate::parse::MemoryMapDesc;
+
+pub mod device_instances;
+pub mod device_types;
+pub mod types;
+
+pub(crate) fn ident(n: impl AsRef<str>) -> Ident {
+    // TODO handle things like CamelCase/snake_case conversion?
+    let s = match n.as_ref() {
+        "(,)" => "Pair",
+        "(,,)" => "Thruple",
```

Comment:
```suggestion
        "(,,)" => "Thruple",
```

Haha, haven't seen that one before! `Triple`?